Samples/resource-owner-password-credential/Angular2/Angualar2-Client-ROPC/app/authorize/authorize-component.ts

```diff
@@ -0,0 +1,202 @@
+﻿import {Component, ViewChild } from '@angular/core'
+import {JwtHelper} from 'angular2-jwt'
+import {Router} from '@angular/router';
+import {authervice} from './authoriza-service';
+import { MODAL_DIRECTIVES, ModalComponent  } from 'ng2-bs3-modal/ng2-bs3-modal';
+declare var System;
+@Component({
+    selector: 'authorize',
+    templateUrl: '/app/authorize/authorize-component.html',
+    directives: [MODAL_DIRECTIVES],
+})
+
+export class authorizeComponent{
+
+    constructor(public jwtHelper: JwtHelper,
+        private _parentRouter: Router,
+        private authentication: authervice) {}
+
+    @ViewChild('myModal')
+    modal: ModalComponent;
+
+    public mclose() {
+        this.modal.close();
+    }
+
+   public mopen() {
+        this.modal.open();
+    }
+
+   public logstatus() {
+       this.isLoggedin = true; 
+   }
+
+    public userDetails: any ;
+    public isLoggedin: boolean;
+    public logMsg: string;
+    public model: logModel;
+    public rmodel: registerModel;
+    public login: boolean;
+    public register: boolean;
+    public user: string;
+    public authdata: any;
+    public hodeModel: boolean = false;
+
+    ngOnInit() {
+       
+        var instance = this;
+         // check if auth key is present
+        if (localStorage.getItem('auth_key')) {
```

Comment:
Actually authserviceis solely responsible for http interaction , it simply calls http and returns mapped data/error. it can not have any method. authorize-component is the only component on whole page which will handle the authorization logic . i have removed logout from user-component as it was unnecessarily there which might have brought that confusion.   and in angular2 there is another way to manage this state using subject etc. but thats choice of the developer. In my application i also have a method which checks if the token has expired so it can get a new access_token from refersh_token and then calls the function , so i really don't think that having everything in one place is good idea. however its your call. bdw for demo you can check this https://swapkart.com demo app